redis/connection.py

```diff
@@ -739,6 +740,10 @@ def _error_message(self, exception):
     def on_connect(self):
         "Initialize the connection, authenticate and select a database"
         self._parser.on_connect(self)
+        if ssl_available:
+            authened = False
```

Comment:
Why do you need this flag?

Reply:
It's for the test_ssl.py in test. In the test_ssl_connection_without_ssl test case, stack overflow occur if authentication fails and the command is sent.

```python
    def test_ssl_connection_without_ssl(self, request):
        ssl_url = request.config.option.redis_ssl_url
        p = urlparse(ssl_url)[1].split(":")
        r = redis.Redis(host=p[0], port=p[1], ssl=False)

        with pytest.raises(ConnectionError) as e:
            r.ping()
            assert "Connection closed by server" in str(e)

Reply:
You don't need it after you catch ResponseError